spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -70,6 +71,12 @@ class DeltaCatalog extends DelegatingCatalogExtension
 
   val spark = SparkSession.active
 
+  val allowedIdentityColumnChanges: Set[Class[_ <: TableChange]] = Set(
+    classOf[UpdateColumnComment],
+    classOf[UpdateColumnPosition],
+    classOf[RenameColumn],
+    classOf[SyncIdentity])
+
```

Comment:
I'm not a fan of doing the matching with the `classOf` stored in a set. I'm paranoid and rather we do the case match directly. (i'll write my suggestion below)